bundle/edu.gemini.shared.util/src/main/scala/edu/gemini/shared/util/DateTimeUtils.scala

```diff
@@ -0,0 +1,86 @@
+  val DefaultZone: ZoneId = ZoneId.systemDefault()
+
+  // Standard UTC formatters.
+  val YYYY_MMM_DD_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MMM-dd").withZone(UTC)
+  val YYYY_MMM_DD_HHMM_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm").withZone(UTC)
+  val YYYY_MMM_DD_HHMMSS_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm:ss").withZone(UTC)
+  val YYYY_MMM_DD_HHMMSS_Z_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm:ss z").withZone(UTC)
+  val HHMMSS_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss").withZone(UTC)
+
+  // Standard non-UTC formatters.
+  val YYYY_MMM_DD_DZ_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MMM-dd").withZone(DefaultZone)
+  val YYYY_MMM_DD_HHMMSS_DZ_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MMM-dd HH:mm:ss").withZone(DefaultZone)
+
+  // Nonstandard formatters.
+  val YYYYMMDD_Formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMdd").withZone(UTC)
+
+  // Create a Comparator<String> from a given DateTimeFormatter.
+  def createComparator(df: DateTimeFormatter): Comparator[String] = Comparator.comparingLong(new ToLongFunction[String] {
+    override def applyAsLong(s: String): Long = ZonedDateTime.parse(s, df).toInstant.toEpochMilli
+  })
+
+
+  // Convenience algorithms.
+  def nearestMinute(date: Date, zone: TimeZone): Date = Date.from(nearestMinute(ZonedDateTime.ofInstant(date.toInstant, zone.toZoneId)).toInstant)
+  def nearestMinute(dt: ZonedDateTime): ZonedDateTime = dt.plus(30, ChronoUnit.SECONDS).truncatedTo(ChronoUnit.MINUTES)
+
+  def startOfDayInMs(ms: Long, zone: ZoneId): Long = {
+    val zdt = Instant.ofEpochMilli(ms - 14.hours.toMillis).atZone(zone)
+    zdt.withHour(14).truncatedTo(ChronoUnit.HOURS).toInstant.toEpochMilli
+  }
+
+  def endOfDayInMs(ms: Long, zone: ZoneId): Long = {
+    val start = startOfDayInMs(ms, zone)
+    val zdt = Instant.ofEpochMilli(start).atZone(zone)
+    zdt.plus(1, ChronoUnit.DAYS).toInstant.toEpochMilli
+  }
+
+  def timeInMs(year: Int, month: Int, dayOfMonth: Int, hours: Int, minutes: Int, seconds: Int, zone: ZoneId): Long =
+    ZonedDateTime.of(year, month, dayOfMonth, hours, minutes, seconds, 0, zone).toInstant.toEpochMilli
+
+  // Java time API doesn't support formatters for Duration, so we have to roll our own.
+  def msToHHMM(span: Long): String = {
+    val neg = span < 0
+    val d   = Math.abs(span).milliseconds
+    val hh  = d.toHours
+    val mm  = d.toMinutes % MinutesPerHour
+    f"${neg ? "-" | ""}%s$hh%02d:$mm%02d"
+  }
+
+  def msToHHMMSS(span: Long): String = {
+    val neg = span < 0
+    val d   = Math.abs(span).milliseconds
+    val hh  = d.toHours
+    val mm  = d.toMinutes % MinutesPerHour
+    val ss  = d.toSeconds % SecondsPerMinute
+    f"${neg ? "-" | ""}%s$hh%02d:$mm%02d:$ss%02d"
+  }
+
+  // Some useful constants.
+  val StartOfDayHour:        Int  = 14
```

Comment:
Let's use this constant on the various places `14` is used

Reply:
Agreed that this is a good idea, and making the change now.

Reply:
Unfortunately, this isn't available to `edu.gemini.spModel.core.Semester`, where I'm guessing `SWITCH_OVER_HOUR = 14` means the same thing.